docs/guide/src/dev/protobuf.md

```diff
@@ -0,0 +1,38 @@
+# Maintaining protobuf specs
+
+The Penumbra project dynamically generates code for interfacing
+with [gRPC]. The following locations within the repository
+are relevant:
+
+  * `proto/proto/penumbra/**/*.proto`, the developer-authored spec files
+  * `proto/src/gen/*.rs`, the generated output files
+  * `tools/proto-compiler/`, the build logic for the generation
+
+## Installing protobuf
+
+Obtain the most recent pre-compiled binary from the [`protoc` website].
+After installing, run `protoc --version` and confirm you're running
+at least `3.21.8` (or newer). Don't install `protoc` from package managers
+such as `apt`, as those versions are often outdated, and will not work
+with Penumbra.
+
+
+## Building protobuf
+
+Switch to the [proto-compiler] directory and run the tool:
+
+
+```shell
+cd tools/proto-compiler
+cargo run
+```
+
+Then run `git status` to determine whether any changes were made.
+The build process is deterministic, so regenerating multiple times
+from the same source files will not change the output.
+
+If changes were made, commit those changes to the repository and PR in.
```

Comment:
It might be worth mentioning that the code in `proto-compiler` crate also "shapes" the Serde implementations of the derived Rust types to have more favorable JSON output (e.g., rendering addresses as Bech32-encoded strings, etc).